pkg/chunk/cache/redis_client.go

```diff
@@ -110,17 +110,29 @@ func (c *RedisClient) MGet(ctx context.Context, keys []string) ([][]byte, error)
 		defer cancel()
 	}
 
-	cmd := c.rdb.MGet(ctx, keys...)
-	if err := cmd.Err(); err != nil {
-		return nil, err
-	}
-
 	ret := make([][]byte, len(keys))
-	for i, val := range cmd.Val() {
-		if val != nil {
-			ret[i] = StringToBytes(val.(string))
+	_, isCluster := c.rdb.(*redis.ClusterClient)
```

Comment:
Just a question to better understand the fix (I'm not an expect on Redis or the Go bindings).

This is working around an issue/limitation in the bindings, correct? Or a limitation in a particular server version? I only ask because my first reaction to this is that checking the type of an interface feels wrong. If it's the only option, then we'll have to live with it of course.

Either way, it might be nice to put a comment here to explain why we have to check the type.

Reply:
Thank you for the review!

I thought the same thing, but I think this method is the simplest. The Redis client is initialized with the `NewUniversalClient` .

https://github.com/cortexproject/cortex/blob/v1.8.0/pkg/chunk/cache/redis_client.go#L70

This is the implementation.

https://github.com/go-redis/redis/blob/v8.8.0/universal.go#L199-L206

`UniversalClient` can take `Client` and `ClusterClient`. Single node or sentinel if `Client` is set. mget is always supported. Redis Cluster configuration if `ClusterClient` is set. mget may not be supported.

The only way to check if the Universal Client is Redis Cluster is by type assertion.

Reply:
Added a comment.